src/dmd/root/array.d

```diff
@@ -1013,7 +1038,7 @@ enum isInputRange(R) =
 
 /// Evaluates to `true` if `func` can be called with a value of `T` and returns `bool`.
 enum isPredicateOf(alias func, T) =
-    is(ReturnType!(typeof((T t) => func(t))) == bool);
+    __traits(compiles, { auto _ = (T t) { if (func(t)) {} }; });
```

Comment:
This change makes sure everything that is convertible to `bool` will be accepted, instead of just `bool`.